charts/opentelemetry-collector/values.yaml

```diff
@@ -8,6 +8,12 @@ fullnameOverride: ""
 # Valid values are "daemonset", "deployment", and "statefulset".
 mode: ""
 
+# Enable the creation of a Service pointing the agent pods with mode = daemonset.
+# By default, Service will be created setting 'internalTrafficPolicy: Local'
+# unless other value is explicitly set on Service.
+# Setting 'internalTrafficPolicy: Cluster' on a daemonset is not recommended
+serviceOnDaemonset: false
```

Comment:
I think since we have to check this value against `mode` in NOTES.txt no matter what I think we can move this under the `service` configuration section.  Please add a check in `NOTES.txt` that will fail the install if serviceOnDaemonset is true and mode != daemonset.

Reply:
Fixed! :)